ex/loqui/lib/loqui/client.ex

```diff
@@ -0,0 +1,358 @@
+defmodule Loqui.Client do
+  defmodule State do
```

Comment:
I'd suggest revisiting that; especially on the server where you're doing the buffering. The add to buffer / flush dance could be expressed much more cleanly as:
```elixir
state 
 |> handle_packets
 |> State.flush_buffer
# in handle_packets 
Frame.reply(...)
|> State.add_to_buffer

# and State.flush
def flush_buffer(%State{socket: sock, buffered_frames: buffer}=state) do 
   :gen_tcp.write(sock, buffer)
   %{state | buffer: []}
end
```